cpp/src/lists/contains.cu

```diff
@@ -255,14 +259,16 @@ void index_of_nested_types(InputIterator input_it,
   // returned to keep the result `table_view` valid.
   [[maybe_unused]] auto const [keys_tview, unused_column] =
     [&]() -> std::pair<table_view, std::unique_ptr<column>> {
-    if constexpr (std::is_same_v<SearchKeyType, cudf::scalar>) {
+    if constexpr (search_key_is_scalar) {
       auto tmp_column       = make_column_from_scalar(search_keys, 1, stream);
-      auto const keys_tview = tmp_column->view();
-      return {table_view{{keys_tview}}, std::move(tmp_column)};
-    } else {
-      return {table_view{{search_keys}}, nullptr};
+      return {table_view{{tmp_column->view()}}, std::move(tmp_column)};
     }
+
+    // Using `if constexpr` again instead of `else` branch to avoid incorrect compiler warning about
```

Comment:
After the removal of the variable above, can we bring this back into an `else` clause?

Reply:
Let's me try that...